backend/courses/filters.py

```diff
@@ -157,6 +157,20 @@ def requirement_filter(queryset, req_ids):
     return queryset.filter(query)
 
 
+def attribute_filter(queryset, attr_codes):
+    """
+    :param queryset: initial Course object queryset
+    :param attr_codes: the attribute codes (ex: WUOM) comma separated
+    :return: filtered queryset
+    """
+    if not attr_codes:
+        return queryset
+    query = Q()
+    for attr_code in attr_codes.split(","):
+        query |= Q(attributes__code=attr_code)
```

Comment:
This is an altered version of the original filter (compare to https://github.com/pennlabs/penn-courses/pull/384/commits/333264471e5923746def47406c2a37bc95eb8689) which does not validate that the `attr_code` exists in the database before adding it to the queryset. As a result, a single invalid `attr_code` will produce an empty queryset whereas it yielded all courses before.